crates/polars-io/src/cloud/options.rs

```
@@ -80,19 +78,11 @@ pub struct CloudOptions {
     pub file_cache_ttl: u64,
     pub(crate) config: Option<CloudConfig>,
     #[cfg(feature = "cloud")]
-    #[cfg_attr(feature = "serde", serde(deserialize_with = "deserialize_or_default"))]
+    /// Note: In most cases you will want to access this via [`CloudOptions::initialized_credential_provider`]
+    /// rather than directly.
     pub(crate) credential_provider: Option<PlCredentialProvider>,
 }
 
-#[cfg(all(feature = "serde", feature = "cloud"))]
-fn deserialize_or_default<'de, D>(deserializer: D) -> Result<Option<PlCredentialProvider>, D::Error>
-where
-    D: Deserializer<'de>,
-{
-    type T = Option<PlCredentialProvider>;
-    T::deserialize(deserializer).or_else(|_| Ok(Default::default()))
```

* This was added by https://github.com/pola-rs/polars/pull/20745 - I'm not sure for the reason. But I've removed it as it was causing deserialization errors to be silently ignored.